neon_homeassistant_skill/__init__.py

```diff
@@ -16,39 +17,112 @@ def chunks(lst, n_len) -> List[list]:
-        )
+                "ovos.phal.plugin.homeassistant.get.light.brightness.response",
+                self.handle_get_light_brightness_response,
+            )
         self.bus.on(
-            "ovos.phal.plugin.homeassistant.set.light.brightness.response",
-            self.handle_set_light_brightness_response,
-        )
+                "ovos.phal.plugin.homeassistant.set.light.brightness.response",
+                self.handle_set_light_brightness_response,
+            )
         self.bus.on(
-            "ovos.phal.plugin.homeassistant.increase.light.brightness.response",
-            self.handle_set_light_brightness_response,
-        )
+                "ovos.phal.plugin.homeassistant.increase.light.brightness.response",
+                self.handle_set_light_brightness_response,
+            )
         self.bus.on(
-            "ovos.phal.plugin.homeassistant.decrease.light.brightness.response",
-            self.handle_set_light_brightness_response,
-        )
+                "ovos.phal.plugin.homeassistant.decrease.light.brightness.response",
+                self.handle_set_light_brightness_response,
+            )
         self.bus.on(
-            "ovos.phal.plugin.homeassistant.get.light.color.response",
-            self.handle_get_light_color_response,
-        )
+                "ovos.phal.plugin.homeassistant.get.light.color.response",
+                self.handle_get_light_color_response,
+            )
         self.bus.on(
-            "ovos.phal.plugin.homeassistant.set.light.color.response",
-            self.handle_set_light_color_response,
-        )
+                "ovos.phal.plugin.homeassistant.set.light.color.response",
+                self.handle_set_light_color_response,
+            )
+        self.bus.on("ovos.phal.plugin.homeassistant.ready", self.on_ready)
+
+    def on_ready(self, message):
+        resp = self.bus.wait_for_response(message.forward("ovos.phal.plugin.homeassistant.check_connected"))
+        self.log.debug(f"Response from HA PHAL plugin: {repr(resp)}")
+        if resp and resp.data.get("connected"):
+            self.log.debug("PHAL plugin connected to HA")
+            self.enable_ha_intents()
+            return True
+        if not resp:
+            self.log.error("Home Assistant PHAL plugin not installed or not running!")
+            self.disable_unused_intents()
+            return False
+
+        self.log.info("PHAL Plugin not connected to HomeAssistant")
+        self.disable_unused_intents()
+        return False
+
+    def disable_unused_intents(self):
+        for intent in self.connected_intents:
+            self.intent_service.remove_intent(intent)
+            try:
+                assert self.intent_service.intent_is_detached(intent) is True
+            except AssertionError:
+                self.log.error(f"Error disabling intent: {intent}")
+
+    def register_intents(self):
+        for intent in self.connected_intents:
+            # TODO: Localization
```

Comment:
Does `self.enable_intent` work here? that should handle all configured languages

Reply:
Not sure, not familiar with it. I'll give it a spin